src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Build/ImplicitlyTriggeredDebugBuildManager.cs

```diff
@@ -59,20 +59,18 @@ protected override async Task InitializeCoreAsync(CancellationToken cancellation
             ErrorHandler.ThrowOnFailure(_solutionBuildManager.AdviseUpdateSolutionEvents3(this, out _cookie3));
 
             _skipAnalyzersForImplicitlyTriggeredBuild = await _options.GetSkipAnalyzersForImplicitlyTriggeredBuildAsync(cancellationToken);
-            _options.RegisterOptionChangedEventHandler(OnOptionChangedAsync);
+            await _options.RegisterOptionChangedEventHandlerAsync(OnOptionChangedAsync);
         }
 
-        protected override Task DisposeCoreAsync(bool initialized)
+        protected override async Task DisposeCoreAsync(bool initialized)
         {
             if (initialized)
             {
                 (_solutionBuildManager as IVsSolutionBuildManager2)?.UnadviseUpdateSolutionEvents(_cookie);
                 _solutionBuildManager!.UnadviseUpdateSolutionEvents3(_cookie3);
 
-                _options.UnregisterOptionChangedEventHandler(OnOptionChangedAsync);
+                await _options.UnregisterOptionChangedEventHandlerAsync(OnOptionChangedAsync);
```

Comment:
This could return the task rather than await it, to avoid a state machine.

Reply:
Done.